### ISSUE TITLE
Feature Request: Toggle Spells in a Spell List with Tags

### ISSUE THREAD

**Issue by uknighted** [ohrrpgce/ohrrpgce#1182]
Currently, you can toggle selections in a battle menu with tags, but that can only fit a certain amount of menu selections. A spell list is much more expansive, however it lacks to ability to toggle attacks with tags. Being able to disable or enable a spell depending on a tag could make battles much more interactive, so I think it could be a worthy addition.

**Comment by rversteegen** [comment]
I agree, this would be an excellent feature. It was on my todo list. It should be fairly easy...

although it means we're going to have to keep the `HeroDef.spell_lists` data in memory because we'll need to loop over it from `tag_updates`. (I want to keep all of HeroDef in memory, but doing so is a significant amount of cleanup). Also, note that there are no script commands for checking/setting whether a spell is learned by level-up, so no commands for tag conditions either won't be a serious omission. Learning a spell from a level up only happens if that slot is empty (`write spell` hasn't been used on it).
